test/hotspot/jtreg/serviceability/jvmti/FollowReferences/FieldIndices/libFieldIndicesTest.cpp

```diff
@@ -0,0 +1,515 @@
+    }
+
+    if (reference_kind == JVMTI_HEAP_REFERENCE_FIELD) {
+        Object* obj = (Object*)(*tag_ptr);
+        Klass* klass = obj->klass;
+        printf("primitiveFieldCallback(JVMTI_HEAP_REFERENCE_FIELD): "
+               "klass=%s, index=%d, type=%c, value=%d\n",
+               klass->name, index,
+               (int)value_type, (int)value.i);
+        if (check_index_bounds(index, klass)) {
+            jint expected_value = obj->field_values[index - klass->interface_field_count];
+            if (int_value != expected_value) {
+                printf("  ERROR: wrong instance value: (%d, expected %d)\n",
+                       (int)int_value, (int)expected_value);
+                test_failed = true;
+            } else {
+                printf("  OK: field %s.%s, value %d\n",
+                       klass->name, get_field_name(klass, index), (int)int_value);
+            }
+        }
+    } else if (reference_kind == JVMTI_HEAP_REFERENCE_STATIC_FIELD) {
+        Klass* klass = (Klass*)(*tag_ptr);
+        printf("primitiveFieldCallback(JVMTI_HEAP_REFERENCE_STATIC_FIELD): "
+               "klass=%s, index=%d, type=%c, value=%d\n",
+               klass->name, index,
+               (int)value_type, (int)value.i);
+        if (check_index_bounds(index, klass)) {
+            jint expected_value = klass->fields[index - klass->interface_field_count].value;
+            if (int_value != expected_value) {
+                printf("  ERROR: wrong static value: (%d, expected %d)\n\n\n",
+                       (int)int_value, (int)expected_value);
+                test_failed = true;
+            } else {
+                printf("  OK: field %s.%s, value %d\n",
+                          klass->name, get_field_name(klass, index), (int)int_value);
+            }
+        }
+    } else {
+        printf("ERROR: unexpected reference_kind in primitiveFieldCallback: %d\n", (int)reference_kind);
+        test_failed = true;
+    }
+
+    fflush(nullptr);
+    return 0;
+}
+
+
+JNIEXPORT void JNICALL
+Java_FieldIndicesTest_prepare(JNIEnv *env, jclass cls, jobject testObj) {
+    Object::explore(env, testObj);
+    fflush(nullptr);
+}
+
+JNIEXPORT void JNICALL
+Java_FieldIndicesTest_test(JNIEnv *env, jclass cls, jobject rootObject) {
+    jvmtiHeapCallbacks heapCallbacks;
+    memset(&heapCallbacks, 0, sizeof(heapCallbacks));
+
+    heapCallbacks.primitive_field_callback = primitiveFieldCallback;
+
+    check_jvmti_status(env,
+        jvmti->FollowReferences(JVMTI_HEAP_FILTER_UNTAGGED, // heap_filter
+                                nullptr,                    // class
+                                rootObject,                 // initial_object
+                                &heapCallbacks,
+                                nullptr),
+        "FollowReferences failed");
+    fflush(nullptr);
+}
+
+JNIEXPORT jboolean JNICALL
+Java_FieldIndicesTest_testFailed(JNIEnv *env, jclass cls) {
+    return test_failed ? JNI_TRUE : JNI_FALSE;
```

Comment:
The indent for native files has to be 2, not 4. Even though there are still some tests with wrong indents I'd suggest for new files to follow this rule.

Reply:
Fixed. Going to integrate the change after sanity testing